pkg/api/api.go

```diff
@@ -52,17 +52,18 @@ type ModelRegistryApi interface {
 
 	// ARTIFACT
 
-	UpsertArtifact(artifact *openapi.Artifact, modelVersionId *string) (*openapi.Artifact, error)
+	UpsertModelVersionArtifact(artifact *openapi.Artifact, modelVersionId string) (*openapi.Artifact, error)
+
+	UpsertArtifact(artifact *openapi.Artifact) (*openapi.Artifact, error)
 
 	GetArtifactById(id string) (*openapi.Artifact, error)
 
 	GetArtifacts(listOptions ListOptions, modelVersionId *string) (*openapi.ArtifactList, error)
 
 	// MODEL ARTIFACT
 
-	// UpsertModelArtifact create a new Artifact or update an Artifact associated to a specific
-	// ModelVersion identified by modelVersionId parameter
-	UpsertModelArtifact(modelArtifact *openapi.ModelArtifact, modelVersionId *string) (*openapi.ModelArtifact, error)
+	// UpsertModelArtifact creates or inserts an Artifact
+	UpsertModelArtifact(modelArtifact *openapi.ModelArtifact) (*openapi.ModelArtifact, error)
```

Comment:
as this PR touches this "Core API" I would seize the chance to: 

1. revise if any changes required in doc: https://github.com/kubeflow/model-registry/blob/main/docs/mr_go_library.md

2. add a disclaimer warning in that doc that it currently expose direct connection to MLMD, and this is potentially subject of change in the future, hence the "mr go library" doc is intended for Developer purposes, while external clients are encouraged to use the REST API

Reply:
oh yes, I forgot about it.